whisper/mailserver/mailserver.go

```diff
@@ -175,7 +174,10 @@ func (s *WMailServer) validateRequest(peerID []byte, request *whisper.Envelope)
 	if len(src)-len(peerID) == 1 {
 		src = src[1:]
 	}
-	if !bytes.Equal(peerID, src) {
+
+	// if you want to check the signature, you can do it here. e.g.:
+	// if !bytes.Equal(peerID, src) {
+	if src == nil {
```

Comment:
is this a temporary solution?

Reply:
no. but it should serve as example for those who want to develope their own mailserver.
we might review this stuff after we swich to libp2p.